File: src/main/kotlin/com/raymanoz/borgy/scale/Scale.kt
package com.raymanoz.borgy.scale

import org.http4k.core.Body
import org.http4k.format.Jackson.auto
import org.http4k.lens.BiDiBodyLens

data class Scale(val name: String, val description: String, val intensities: List<Intensity>) {
    companion object {
        val lens: BiDiBodyLens<Scale> = Body.auto<Scale>().toLens()
        val scalesLens: BiDiBodyLens<List<Scale>> = Body.auto<List<Scale>>().toLens()
    }
}
data class Intensity(val number: Double, val label: String)

File: src/main/react/src/pages/Scale.ts
export interface Intensity {
    number: number;
    label: string;
}

export interface Scale {
    name: string;
    description: string;
    intensities: Intensity[];
}

export interface TrialSummary {
    name: string;
    state: string;
}

export type Scales = Scale[];
export type Trials = TrialSummary[];
